develop

SvetlanaZakharchuk opened the pull request.

SvetlanaZakharchuk committed:
develop

SvetlanaZakharchuk committed:
Update AnswersTest.java

SvetlanaZakharchuk committed:
Update TestFormForSamakatOrder.java

SvetlanaZakharchuk committed:
Update CheckReadyOrder.java

SvetlanaZakharchuk committed:
Update MainPage.java

SvetlanaZakharchuk committed:
Update OpenOrderForm.java

SvetlanaZakharchuk committed:
Update PersonInfo.java

SvetlanaZakharchuk committed:
Update SamokatInfo.java

pomegranate282922 commented on src/main/java/edu/praktikum/samokat/CheckReadyOrder.java:
```diff
@@ -0,0 +1,37 @@
+
+    //нажимаем кнопку "Да"
+    public void clickButtonYes(){
+        driver.findElement(buttonYes).isEnabled();
```
Эта строчка ничего не делает.
Нужно исправить

pomegranate282922 commented on src/main/java/edu/praktikum/samokat/MainPage.java:
```diff
@@ -0,0 +1,113 @@
+    }
+    //открываем браузер
+    public void openUrl(){
+        driver.get("https://qa-scooter.praktikum-services.ru/");
```
Урлы лучше вынести как константы и переиспользовать

pomegranate282922 commented on src/main/java/edu/praktikum/samokat/PersonInfo.java:
```diff
@@ -0,0 +1,74 @@
+        driver.findElement(buttonNext).click();
+    }
+    //собиарем все методы для заполнения полей в один
+    public void FillPersonInfo(String name, String surname, String adress, String phoneNumber){
```
Все методы именуем начиная со строчной буквы
Нужно исправить везде

pomegranate282922 commented on src/test/java/edu/praktikum/samokat/AnswersTest.java:
```diff
@@ -0,0 +1,64 @@
+
+
+    @Before
+    public void SetUp() {
```
Со строчной буквы 
Нужно исправить

pomegranate282922 commented on src/test/java/edu/praktikum/samokat/AnswersTest.java:
```diff
@@ -0,0 +1,64 @@
+        Assert.assertEquals(mainPage.getAnswerSix(), ANSWER6);
+        Assert.assertEquals(mainPage.getAnswerSeven(), ANSWER7);
+        Assert.assertEquals(mainPage.getAnswerEight(), ANSWER8);
+
```
Все 8 проверок должны быть независимы. 
нужно исправить

pomegranate282922 commented on src/test/java/edu/praktikum/samokat/TestFormForSamakatOrder.java:
```diff
@@ -0,0 +1,116 @@
+        CheckReadyOrder checkReadyOrder = new CheckReadyOrder(driver);
+        checkReadyOrder.clickButtonYes();
+        //проверяем, что заказ сформирован
+        assertEquals(checkReadyOrder.isOrderReadyOpen(), "Заказ оформлен");
```
Сначала должно идти ожидаемое значение, потом реальное.
Нужно исправить везде

pomegranate282922 commented on src/test/java/edu/praktikum/samokat/TestFormForSamakatOrder.java:
```diff
@@ -0,0 +1,116 @@
+    }
+
+    @Before
+    public void SetUp() {
```
Общие начала и концы тестов удобно вынести в общий класс предок

pomegranate282922 commented on src/test/java/edu/praktikum/samokat/TestFormForSamakatOrder.java:
```diff
@@ -0,0 +1,116 @@
+        CheckReadyOrder checkReadyOrder = new CheckReadyOrder(driver);
+        checkReadyOrder.clickButtonYes();
+        //проверяем, что заказ оформлен
+        assertEquals(checkReadyOrder.isOrderReadyOpen(), "Заказ оформлен");
```
Нужно обязательно убедиться, что в браузерах отличных от хрома эти тесты зеленые.

pomegranate282922 commented on pom.xml:
```diff
@@ -0,0 +1,17 @@
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+</project>
```
Нет зависимостей. Мавен Проект не компилируется 
Нужно исправить

SvetlanaZakharchuk replied:
Добавила нужные зависимости

pomegranate282922 commented on src/main/java/edu/praktikum/samokat/PersonInfo.java:
```diff
@@ -0,0 +1,74 @@
+        inputPhone(phoneNumber);
+    }
+}
+
```
Отлично! Удачно подобраны локаторы

pomegranate282922 reviewed (commented):
Работа хорошая! Реализованы все тесты . Уместно использована параметризация. 
но есть и важные замечания.
1. В помнике нет зависимостей 
2. Тесты на faq должны дать независимыми друг от друга. Нужно использовать параметризацию или сделать 8 отдельных тестов 
3. Нейминг 
4. Нужно убедиться, что тесты на создание заказа НЕ в хроме зеленые 
5. Остальное по мелочи

SvetlanaZakharchuk committed:
Update pom.xml

добавила нужные зависимости

SvetlanaZakharchuk committed:
Update CheckReadyOrder.java

Убрала лишнюю строку

SvetlanaZakharchuk committed:
Update AnswersTest.java

Сделала url константой и добавила тест на проверку каждого вопроса

SvetlanaZakharchuk committed:
Update TestFormForSamakatOrder.java

добавила константы и исправила наименования методов, вынесла общие части для тестов в отдельный класс

SvetlanaZakharchuk committed:
Update MainPage.java

изменила адрес страницы на переменную.

SvetlanaZakharchuk committed:
Update OpenOrderForm.java

исправила url

SvetlanaZakharchuk committed:
Update PersonInfo.java

исправила название метода со строчной буквы

SvetlanaZakharchuk committed:
Update TestFormForSamakatOrder.java

поменяла в asserte ожидаемое и фактическое

SvetlanaZakharchuk committed:
Add files via upload

тестовый класс-предок, в который вынесены общие методы для тестовых классов

pomegranate282922 commented on src/test/java/edu/praktikum/samokat/AnswersTest.java:
```diff
@@ -0,0 +1,108 @@
+        options.addArguments("--remote-allow-origins=*");
+        driver = new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    }
```
Закомментированный код нужно удалять

pomegranate282922 reviewed (approved):
Ок, зачет!
